### PR TITLE
Focal plane coords

### PR DESCRIPTION
**Pull request by skhrg**
Module to generate focal plane from physical optics, this is just the LAT for now. I need to touch base with SAT optics folks about SAT optics again (but it should be simpler than the LAT).

One question I did have was if the corotator angle would be something that will be nominally loaded into the axis manager for LAT data (I assume yes)? And if we had a standardized name for it so I can have this code pull it from the aman by default.

### CODE REVIEW

**Comment by Zhilei** [comment]
I am wondering whether @laurensaunders has any inputs in the corotator angle. She is the most experienced with the ACU work.

File: sotodlib/coords/focal_plane.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+import logging
+import numpy as np
+from scipy.interpolate import interp2d
+from scipy.spatial.transform import Rotation as R
+from sotodlib import core
+
+logger = logging.getLogger(__name__)
+
+# Dictionary of zemax tube layout.
+# +ve x is to the right as seen from back of the cryostat *need to check*
```
Comment (Zhilei) [inline sotodlib/coords/focal_plane.py:16]: I am not sure about this layout, here is what I keep the record when I looked into this
https://docs.google.com/spreadsheets/d/1od2IfJo6Czt9ExH7wDADbaRPL77qi270XpjnF0f5c4U/edit#gid=0

**Review by Zhilei — CHANGES_REQUESTED** [review]
Thanks for the nice PR! I would suggest we check the layout of the optics tubes before we proceed to merge.

File: sotodlib/coords/focal_plane.py (hunk 1 above)
Comment (skhrg) [reply sotodlib/coords/focal_plane.py:16]: Interesting, there is an off by one error that I introduced for indexing purposes but even with that many of these are inconsistent. I will check with @sdicker to see what he thinks.

**Comment by skhrg** [comment]
Added in SAT optics and some performance improvements. I'd like to get this merged soonish since I want this done for `finalize-focal-plane`. @Zhilei what do you think about merging for now and when I get back to NA I will double check the OT indexing and if there is a problem put in another PR with the fix.

**Comment by tskisner** [comment]
Just a note that we should call these functions in the [stub that already exists for it](https://github.com/simonsobs/sotodlib/blob/2cf83c7b06a1e175ebd3fd75bf11dd26c4540b5b/sotodlib/sim_hardware.py#L47-L66) when simulating focalplane layout.

Also, I previously gathered my notes on [LAT optics tube configuration](https://docs.google.com/document/d/1RjcPj8zEsid-Q2USwBT_g2ww5LmmTkQUwoZDIhniAKg/edit?usp=sharing) here if it helps.

**Comment by skhrg** [comment]
Thanks Ted, I think your notes agree with what I got from Simon. If we did get the mapping wrong it should (I think) be obvious the first time we compare this against measured pointing so I am not super worried about getting it 100% correct right now.
I will fill out that stub when I get back to NA, may put that is a second PR since I don't like how old this branch is getting.

**Comment by skhrg** [comment]
Also, I think we should gather all these notes on tube layout and the optics somewhere on the confluence so its in a central location. Any oppositions/ideas where that belongs?

**Comment by sdicker** [comment]
For that matter all the optical design files are on the old wiki - I claim
nicely organized - maybe a link from such a confluence page could be in
order.

**Comment by skhrg** [comment]
All the pages from the old wiki are converted to confluence pages. I can migrate it to the main space and then we can add in any information from Ted and Zhilei's notes that is missing. Maybe we put this in a LAT_Optics page under the LAT tree?

**Review by Zhilei — APPROVED** [review]
Let's move forward!

**Pull request merged**